Netdata Agent: Backup restore documentation (#17006)
netdata/netdata: docs/maintenance/backup-restore.md

# Backing up a Netdata Agent

## Introduction

When preparing to backup a Netdata Agent it is worth considering that there are different kinds of data that you may wish to backup independently or all together:

| Data type           | Description                                          | Location                                                                                                                |
|---------------------|------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------|

> | Agent configuration | Files controlling configuration of the Netdata Agent | [config directory](https://github.com/netdata/netdata/blob/master/docs/configure/nodes.md#the-netdata-config-directory) |

Reviewer: ```suggestion
| Agent configuration | Files controlling configuration of the Netdata Agent | [config directory](https://github.com/netdata/netdata/blob/master/docs/netdata-agent/configuration.md) |
```

   > **Note**  
   > The specific paths may vary depending upon installation method, Operating System and whether it is a Docker/Kubernetes deployment.

2. It is recommended that you [stop the Netdata Agent](https://github.com/netdata/netdata/blob/master/docs/configure/start-stop-restart.md) when backing up the Metrics/database files.  
   Backing up the Agent configuration and Identity folders is straight-forward as they should not be changing very frequently.

3. Using a backup tool such as `tar` you will need to run the backup as _root_ or as the _netdata_ user in order to access all the files in the directories.

  ```
  sudo tar -cvpzf netdata_backup.tar.gz /etc/netdata/ /var/cache/netdata /var/lib/netdata
  ```
  
  Stopping the Netdata agent is mostly required in order to back up the _database files_ of the Netdata Agent.
  
  If you wish to minimize the gap in metrics caused by stopping the Netdata Agent, then you could have a backup job or script that uses the following sequence:
  
- Backup the Agent configuration Identity directories
- Stop the Netdata service
- Backup up up the database files
- Restart the netdata agent.

### Restoring Netdata

> 1. Ensure that the Netdata agent is installed and is [stopped](https://github.com/netdata/netdata/blob/master/docs/configure/start-stop-restart.md)

Reviewer: ```suggestion
1. Ensure that the Netdata agent is installed and is [stopped](https://github.com/netdata/netdata/blob/master/packaging/installer/README.md#maintaining-a-netdata-agent-installation)
```

>    If you plan to deploy the Agent and restore a backup on top of it, then you might find it helpful to use the [`--dont-start-it`](https://learn.netdata.cloud/docs/installing/one-line-installer-for-all-linux-systems#other-options) option upon installation.

Reviewer: ```suggestion
   If you plan to deploy the Agent and restore a backup on top of it, then you might find it helpful to use the [`--dont-start-it`](https://github.com/netdata/netdata/blob/master/packaging/installer/methods/kickstart.md#other-options) option upon installation.
```

   ```
   wget -O /tmp/netdata-kickstart.sh https://get.netdata.cloud/kickstart.sh && sh /tmp/netdata-kickstart.sh --dont-start-it
   ```
  
    > **Note**
    > If you are going to restore the database files then you should first ensure that the Metrics directory is empty.
    > 
    > ```
    > sudo rm -Rf /var/cache/netdata
    > ```

2. Restore the backup from the archive

    ```
    sudo tar -xvpzf /path/to/netdata_backup.tar.gz -C /
    ```

> 3. [Start the Netdata agent](https://github.com/netdata/netdata/blob/master/docs/configure/start-stop-restart.md)

Reviewer: ```suggestion
3. [Start the Netdata agent](https://github.com/netdata/netdata/blob/master/packaging/installer/README.md#maintaining-a-netdata-agent-installation)
```

> <!--
> title: "Backing up Netdata"
> description: "Guide on how to backup and restore Netdata."
> custom_edit_url: "https://github.com/netdata/netdata/edit/master/docs/maintenance/backup-restore.md"
> sidebar_label: "Notify"
> learn_status: "Published"
> learn_rel_path: "Integrations/Notify"
> -->

Reviewer: This is not needed, but it doesn't break anything either, remnants of an older time :smile: 
```suggestion
```

# Introduction
When preparing to backup Netdata it is worth considering that there are different sorts of data that you may wish to backup independently or all together:

| Data type      | Description | Location |
| ----------- | ----------- | ----------- |

> | Agent configuration| Files controlling configuration of the Netdata agent | /etc/netdata/

Reviewer: ```suggestion
| Agent configuration| Files controlling configuration of the Netdata Agent | [config directory](https://github.com/netdata/netdata/blob/master/docs/configure/nodes.md#the-netdata-config-directory)
```

| Metrics   | Database files | /var/cache/netdata |
| Identity   | Cloud-claim, API key and some other files | /var/lib/netdata |

> <br>

Reviewer: ```suggestion
```

If you do two line breaks I think it doesn't need a br!

> Note: Only you can decide how to perform your backups. The following examples are provided as a guide only.

Reviewer: ```suggestion
> ###Note
> Only you can decide how to perform your backups. The following examples are provided as a guide only.
```

> <br>

Reviewer: ```suggestion
```
